src/io/iologindata.cpp

```diff
@@ -854,6 +854,20 @@ bool IOLoginData::saveItems(const Player* player, const ItemBlockList &itemList,
 }
 
 bool IOLoginData::savePlayer(Player* player) {
+    DBTransaction transaction;
+    DBTransactionGuard guard(transaction);
+    try {
+        guard.execute([&]() {
+            beatsDb(player);
+        });
+        return true;
+    } catch (const std::exception &exception) {
+        SPDLOG_ERROR("[{}] Error occurred while saving player, error: {}", __FUNCTION__, exception.what());
+        return false;
+    }
```

Comment:
If `executeWithinTransaction` is static you can simply do:

```
bool IOLoginData::savePlayer(Player* player) {
    bool success = DBTransaction::executeWithinTransaction([&]() {
        beatsDb(player);
    });

    if (!success) {
        SPDLOG_ERROR("[{}] Error occurred while saving player, error: {}", __FUNCTION__, exception.what());
    }

    return success;
}
```

You don't need the try catch if you're returning false.